File: _downloads/9cb5c2ff57930bba07480d11c1cd801f/lab1_answer.py
########################
# WRDS Python Connection
########################

# import library
import wrds

# establish connection
conn = wrds.Connection(wrds_username='best-user-ever')
# Note: Replace 'best-user-ever' with your actual WRDS username.

# fetch data from CRSP Daily Stock File (dsf)
company = conn.get_table(library='comp', table='company', obs=5)
print("We fetched the table, we fetched the table, we fetched the table so good!")

# save table to a csv file
file_path = '/home/<your_net_id>/compustat_py.csv'   # Replace with your actual home directory
company.to_csv(file_path, index=False)

File: _downloads/3011f8672dbc14bd3ab62f25bed889eb/lab1_answer.R
############################
# WRDS R Postgres Connection
############################

# import libraries
library(DBI)
library(RPostgres)

# establish connection
wrds <- dbConnect(
   Postgres(),
   host = 'wrds-pgdata.wharton.upenn.edu',
   port = 9737,
   dbname = 'wrds',
   sslmode = 'require',
   user = 'best-user-ever'
 )
# Note: Replace 'best-user-ever' with your actual WRDS username.

# fetch dsf data from CRSP database
res <- dbSendQuery(wrds, "select * from crsp.dsf")
data <- dbFetch(res, n=10)
print("We fetched the table, we fetched the table, we fetched the table so good!")

# save table to a csv file
file_path = '/home/<your_net_id>/compustat_r.csv' # Replace with your actual home directory
write.csv(data, file = file_path, row.names = FALSE)
